### DISCUSSION TITLE
Struggling with fan speed between infill bridge, bridge, and short layers

### DISCUSSION THREAD

**Discussion by kevenwyld** [supermerill/SuperSlicer#1601] [Q&A] [answered]
I'm struggling with setting fan speeds correctly without getting some weird results.

Ideally, and in it's most basic form, I'd like to enforce 10% cooling for infill bridge and pretty much default/auto for everything else.

Unfortunately it seems I'm not able to achieve this without adjusting the `fan_below_layer_time` value very very low which has some unusual results.  I have two parts that I'm printing sequentially, one with a fair bit of infill bridge, and the other with an overhang bridge.  This illustrates my problem quite well.  Please see pictures and captions below:

![Screenshot_2021-10-01_20-15-17](https://user-images.githubusercontent.com/11936535/135700754-bf290ef0-bcfd-4ce2-b11f-19de79853a71.jpg)
Here is the configuration I've landed on.

![Screenshot_2021-10-01_20-16-13](https://user-images.githubusercontent.com/11936535/135700763-61fd9829-2a00-42fc-be6d-087001ff8b93.jpg)
In the infill bridge I've got a good fan speed.  This could be zero, but ideally I'd like some control for a non-zero value so I landed here.

![Screenshot_2021-10-01_20-16-37](https://user-images.githubusercontent.com/11936535/135700804-31e7f570-f7b5-460d-af62-2f2e07270ba3.jpg)
The bridge section of the part is also good, the fan speed is high for the overhanging bridge, as desired.

![Screenshot_2021-10-01_20-17-02](https://user-images.githubusercontent.com/11936535/135700844-fc185a4b-5c03-4d6e-b4d9-d88352caa625.jpg)
Here's a better view of the layers above the bridge in the top part, and where the problem starts.

So the issue is that in order to achieve the current fan speeds I had to set the default fan speed to zero.  If I increase it, the fan speed for the first part on the infill bridge goes to maximum (due to short layer height scaling).  If I set it to zero, then much of the infill above and surrounding the real overhanging bridge gets almost no fan.  This is an issue on this part since I'd like to print the arch unsupported for good surface quality, but if there's no fan it will sag.

Am I doing this wrong?  I did see https://github.com/supermerill/SuperSlicer/issues/1414 which seems like it might be related but there hasn't been a release since that last commit and I haven't tried building from source yet.  Maybe it would fix this issue but I'm not sure.

Thanks in advance for any advice!

**Comment by supermerill** [comment]
> I did see #1414 which seems like it might be related but there hasn't been a release since that last commit and I haven't tried building from source yet. Maybe it would fix this issue but I'm not sure.

to get the latest nightly build, go to the "action" menu and select the latest lien with your os and with the tag "nightly"

**Reply by kevenwyld** [reply]
Thanks! I didn't realize you had nighties.

So even with the latest nightly including the commit I referenced before this still doesn't seem right to me.  I'm not a 3d printing expert though so maybe I just don't understand why things are arranged the way they are.

It seems to me that being able to adjust infill bridge fan speed separate from overhanging bridges is essential and that functionality does exist, but I can't manage to set the infill bridge fan speed lower than overhanging bridges without it also impacting the infill it self.  This seems to be because the `bridge_internal_fan_speed` parameter does not lower the fan speed, it will only increase it.  See example images/videos and captions below:

![Screenshot_2021-10-02_11-36-47](https://user-images.githubusercontent.com/11936535/135726747-2d7885cb-ac91-4bef-a6af-e9d3c90dd06b.jpg)
Lets say I want a default fan speed of 35%, 40% for tops and external perimeters, 70% (max because my fan is overpowered) for overhanging bridges, and 10% for internal bridges.  I've disabled the `fan_below_layer_time` parameter for this example so things are easier for me to follow.


https://user-images.githubusercontent.com/11936535/135732314-5089e949-c403-43e8-81f6-24399f07f579.mp4

The internal bridges are only as low as the default fan speed.  They can't be made lower than the infill speed (which is the default speed)

I would think that wanting the infill bridges to have lower fan speed than everything else was a relatively normal thing.  Maybe I'm just thinking about this all wrong though and I should never need that.

Also thanks again for your help and all your effort towards developing this software!

**Reply by kevenwyld** [reply]
I should clarify that my experience with printing this part and others has been that I either get sagging infill (because it's too hot), or curling infill bridges because they are too cool.  I find that I'm adjusting these parameters to meet the needs of every part I print instead of finding a combination that works well with the filament I'm using in general and leaving it that way.

EDIT: grammar

**Reply by kevenwyld** [reply]
So I did a bit more testing and reading.  I think I'm going about this the wrong way and that's why I'm bumping up  against this "limitation" which isn't really that limiting after all.

I think I need to better control internal bridge overlapping, width, and flow with more cooling to keep them from deforming like they were on my parts instead of less cooling and more overlap.

**Reply by supermerill** [reply]
bridge colling can't go below default colling, because why?
When the cooling increase from "short layer time", it can go over the bridge fan speed, and so I make it able to follow, as I don't see the benefit to keep it lower. For bridges, we want quicker cooling.

top fan and external fan are different, they are absolute as it's for aesthetic purpose, so the filament gets a certain finish, whatever the layer time is.

**Reply by kevenwyld** [reply]
Thanks.  Yeah you're right it doesn't make sense to go below the default speed.  I solved my problem by adjusting the bridge overlap which seems to have been the cause all along.  I was looking for a solution in the wrong place.

I can't really mark replies to a thread as an answer so I guess I'll just leave as is unless you prefer something else.

**Reply by supermerill** [reply]
There is an 'Unanswered' tag on top. So maybe there is a "marked as answered" somewhere.

**Comment by kevenwyld** [comment, marked as answer]
> bridge colling can't go below default colling, because why? When the cooling increase from "short layer time", it can go over the bridge fan speed, and so I make it able to follow, as I don't see the benefit to keep it lower. For bridges, we want quicker cooling.
> 
> top fan and external fan are different, they are absolute as it's for aesthetic purpose, so the filament gets a certain finish, whatever the layer time is.

supermerill answered this in a thread reply.
